diff --git a/node_modules/@voltagent/server-hono/dist/index.js b/node_modules/@voltagent/server-hono/dist/index.js
index 7bcab21..cb8df60 100644
--- a/node_modules/@voltagent/server-hono/dist/index.js
+++ b/node_modules/@voltagent/server-hono/dist/index.js
@@ -1784,7 +1784,9 @@ var createRoute = /* @__PURE__ */ __name((routeConfig) => {
   };
   return Object.defineProperty(route, "getRoutingPath", { enumerable: false });
 }, "createRoute");
-extendZodWithOpenApi(import_zod.z);
+if (import_zod.z.core?.version?.major !== 4) {
+  extendZodWithOpenApi(import_zod.z);
+}
 function addBasePathToDocument(document, basePath) {
   const updatedPaths = {};
   Object.keys(document.paths).forEach((path) => {
diff --git a/node_modules/@voltagent/server-hono/dist/index.mjs b/node_modules/@voltagent/server-hono/dist/index.mjs
index d6098ed..c0cf1d2 100644
--- a/node_modules/@voltagent/server-hono/dist/index.mjs
+++ b/node_modules/@voltagent/server-hono/dist/index.mjs
@@ -1809,7 +1809,9 @@ var createRoute = /* @__PURE__ */ __name((routeConfig) => {
   };
   return Object.defineProperty(route, "getRoutingPath", { enumerable: false });
 }, "createRoute");
-extendZodWithOpenApi(z);
+if (z.core?.version?.major !== 4) {
+  extendZodWithOpenApi(z);
+}
 function addBasePathToDocument(document, basePath) {
   const updatedPaths = {};
   Object.keys(document.paths).forEach((path) => {
